cpp/ql/lib/ext/oleauto.model.yml

```diff
@@ -0,0 +1,8 @@
+extensions:
+  - addsTo:
+      pack: codeql/cpp-all
+      extensible: summaryModel
+    data: # namespace, type, subtypes, name, signature, ext, input, output, kind, provenance
+      - ["", "", False, "SysAllocString", "", "", "Argument[*0]", "ReturnValue[*]", "value", "manual"]
+      - ["", "", False, "SysAllocStringByteLen", "", "", "Argument[*0]", "ReturnValue[*]", "value", "manual"]
+      - ["", "", False, "SysAllocStringLen", "", "", "Argument[*0]", "ReturnValue[*]", "value", "manual"]
```

Comment:
These two presumably can cut the tail of the string passed in as an argument (when the passed length is too short). I assume we still consider that value flow (although part of the data has gone missing)?

Reply:
That's a good point. We model that somewhat inconsistency across different models. For example, [strncpy](https://github.com/github/codeql/blob/039b2ecf9b8bceb8c29fc22f0dee57c1482fb541/cpp/ql/lib/semmle/code/cpp/models/implementations/Strcpy.qll#L118-L119) is currently modeled as taint because the length argument could mean that we only did a partial copy. In practice, I have a feeling that most of these length-bounded functions are used simply as defensive programming and do provide full copies in most cases. So nowadays I tend to lean towards modeling them as value-preserving.

Reply:
`SysAllocString[Byte]Len` are used when the input string is not null-terminated. `BSTR`s store their length explicitly, so it's really more that `SysAllocString(s)` is a shorthand for `SysAllocStringLen(wcslen(s))`. It's the same as the constructors for `std::basic_string`.